drivers/peci/peci_handlers.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/peci.h>
+#include <syscall_handler.h>
+
+static inline int z_vrfy_peci_config(struct device *dev, u32_t bitrate)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, config));
+
+	return z_impl_peci_config((struct device *)dev, (u16_t)bitrate);
+}
+#include <syscalls/peci_config_mrsh.c>
+
+static inline int z_vrfy_peci_enable(struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, enable));
+
+	return z_impl_peci_enable((struct device *)dev);
+}
+#include <syscalls/peci_enable_mrsh.c>
+
+static inline int z_vrfy_peci_disable(struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, disable));
+
+	return z_impl_peci_disable((struct device *)dev);
+}
+#include <syscalls/peci_disable_mrsh.c>
+
+static inline int z_vrfy_peci_transfer(struct device *dev,
+				       const struct peci_msg *msg)
+{
+	struct peci_msg msg_copy;
+	struct peci_buf tx_buf_copy;
+	struct peci_buf rx_buf_copy;
+	const struct peci_buf *tx_buf = msg->tx_buffer;
+	const struct peci_buf *rx_buf = msg->rx_buffer;
+
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, transfer));
+
+	/* Use a local copy to avoid switcheroo attacks. */
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((const struct peci_msg *)msg,
+				      sizeof(struct peci_msg)));
+	Z_OOPS(z_user_from_copy(&msg_copy, msg, sizeof(msg_copy)));
+
+	/* Same with member variables */
+	if (tx_buf) {
+		Z_OOPS(Z_SYSCALL_MEMORY_READ(tx_buf, sizeof(tx_buf_copy)));
+		z_user_from_copy(tx_buf_copy, tx_buf, sizeof(tx_buf_copy));
+		z_user_from_copy(tx_buf_copy.buf, tx_buf->buf, tx_buf->len);
+	} else {
+		memset(&tx_buf_copy, 0, sizeof(tx_buf_copy));
+	}
+
+	if (rx_buf) {
+		Z_OOPS(Z_SYSCALL_MEMORY_READ(rx_buf, sizeof(rx_buf_copy)));
+		z_user_to_copy(rx_buf_copy, rx_buf, sizeof(rx_buf_copy));
+	} else {
+		memset(&rx_buf_copy, 0, sizeof(rx_buf_copy));
+	}
+
+	/* Update pointers from stack copy */
+	msg_copy.tx_buffer = &tx_buf_copy;
+	msg_copy.rx_buffer = &rx_buf_copy;
+
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((msg->tx_buffer->buf,
+				     (msg->tx_buffer->len));
+
+	/* Rx buffer should be able to hold expected data plus fcs byte */
```

Comment:
I really suggest you reconsider this design. Just make 'len' the total size of the data.
As it stands, this is incorrect. You don't account for overflow here.

Reply:
Removing peci_handlers for now. To be added in separate PR